modules/vaos/app/services/vaos/v2/appointments_service.rb

```diff
@@ -714,6 +717,35 @@ def set_cancellable_false(appointment)
         appointment[:cancellable] = false
       end
 
+      def set_modality(appointment)
+        modality = nil
+        if appointment[:service_type] == 'covid'
+          modality = 'vaInPersonVaccine'
+        elsif appointment.dig(:service_category, 0, :text) == 'COMPENSATION & PENSION'
+          modality = 'claimExamAppointment'
+        elsif appointment[:kind] == 'clinic'
+          modality = 'vaInPerson'
+        elsif appointment[:kind] == 'telehealth'
+          modality = telehealth_modality(appointment)
+        elsif appointment[:kind] == 'phone'
+          modality = 'vaPhone'
+        elsif appointment[:kind] == 'cc' && !appointment[:start].nil?
+          modality = 'communityCare'
+        end
+
+        appointment[:modality] = modality
+      end
+
+      def telehealth_modality(appointment)
+        if !appointment.dig(:telehealth, :atlas).nil?
+          'vaVideoCareAtAnAtlasLocation'
+        elsif %w[CLINIC_BASED STORE_FORWARD].include?(appointment.dig(:telehealth, :vvs_kind))
+          'vaVideoCareAtAVaLocation'
+        elsif appointment.dig(:telehealth, :vvs_kind) == 'MOBILE_ANY/ADHOC'
```

Comment:
Hmm I'm not sure if this is right. 

I think this might be based on [an outdated helper method](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/applications/vaos/services/appointment/index.js#L489-L494). Instead, the most up to date logic is [here](https://github.com/department-of-veterans-affairs/vets-website/blob/e5a7c7214041d1dc3e7c9b771d77d43f3e5dba14/src/applications/vaos/appointment-list/redux/selectors.js#L473-L480). 

cc @simiadebowale to double check if that makes sense since he investigated the original spike ticket.

Reply:
It looks like we're implementing the requested logic, but if @simiadebowale wants us to update this, we definitely can do that as well

Reply:
@devin-mccurdy Logic for Telehealth is accurate, thank you